### ISSUE TITLE
[Request] Backport "protocol" option to single-local configuration 

### ISSUE THREAD

**Issue by spyophobia** [shadowsocks/shadowsocks-rust#485]
## TLDR;
After the breaking changes introduced by v1.10, it's now impossible to specify the protocol for `sslocal` when using a single-instance config file.

## Before update
Previously on v1.9, this is how I used `sslocal`:
```bash
sslocal -c config.json --acl my_acl.acl --protocol http
```
I cannot use the default `socks` protocol because my particular client does not support SOCKS proxy.

## After update to v1.10
Now it is no longer possible to use `--protocol http` option unless I specify my server in args using `--local-addr`.

I checked documentation and found that it's now possible to specify protocol in config file, but it's only available when using "extended multiple local configuration". But I don't need multiple local instances; all I want is a different protocol.

## My Request
Please consider making this possible:
```
{
  "servers": [
    {
      "address": "my_server.example.org",
      // ...
    }
  ],
  "local_port": 1080,
  "local_address": "0.0.0.0",
  "protocol": "http"
}
```

Thanks in advance. Much appreciated.

**Comment by zonyitoo** [comment]
Well, you can still use multiple local configuration with just one local:

```
{
    "locals": [
        {
            "local_port": 1080,
            "local_address": "0.0.0.0",
            "protocol": "http"
        }
    ]
}
```

**Comment by spyophobia** [comment]
I would love to do this myself and just start a PR considering how simple this is, but unfortunately I've gotten really busy lately. So here I am asking others a favour instead 😳. Sorry about that.

**Comment by spyophobia** [comment]
Thanks for the quick commit.

I just wish to clarify that I wasn't trying to push you when I talked about starting a PR myself 😅. I don't want to be misinterpreted as being a dick.
